File: python/aws-rekognition/rekognition_compare_faces/local_image_input.py
#!/usr/bin/env python

"""
Sample code to run compare faces api with local image as input
"""

import requests
import os
import sys
import base64


def imageToBase64(srcPath):
    with open(srcPath, 'rb') as image:
        b64Img = base64.b64encode(image.read()).decode('utf-8')
    return b64Img


def infer():
    # Get the API key for invoking Tiyaro API
    apiKey = os.getenv("TIYARO_API_KEY")
    if apiKey is None:
        print("Please set TIYARO_API_KEY environment variable. You can generate your API key from here - https://console.tiyaro.ai/apikeys")
        sys.exit(1)

    # API endpoint
    url = "https://api.tiyaro.ai/v1/ent/aws/1/rekognition_compare_faces"

    img1Path = "../../../testdata/tom1.jpg"
    img1B64 = imageToBase64(img1Path)

    img2Path = "../../../testdata/tom2.jpg"
    img2B64 = imageToBase64(img2Path)

    payload = {
        "SourceImage": {"Bytes": img1B64},
        "TargetImage": {"Bytes": img2B64}
    }

    headers = {
        "accept": "*/*",
        "content-type": "application/json",
        "authorization": f"Bearer {apiKey}"
    }

    response = requests.request("POST", url, json=payload, headers=headers)
    # Check for errors
    response.raise_for_status()

    # Inference response
    print(response.text)


if __name__ == "__main__":
    infer()

File: java/src/main/java/ai/tiyaro/samples/recognition_classes/recognition_compare_faces/LocalImageInput.java
package ai.tiyaro.samples.recognition_classes.recognition_compare_faces;

import org.apache.commons.io.FileUtils;
import org.json.JSONObject;

import java.io.File;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpRequest.BodyPublishers;
import java.net.http.HttpResponse.BodyHandlers;
import java.util.Base64;
import java.util.Collections;

// Sample code to run rekognition compare faces api with local image as input
public class LocalImageInput {

  public String imageToBase64(String filePath) throws Exception {
    File file = new File(LocalImageInput.class.getClassLoader().getResource(filePath).toURI());
    var fileContent = FileUtils.readFileToByteArray(file);
    String encodedBytesString = Base64.getEncoder().encodeToString(fileContent);
    return encodedBytesString;
  }

  public void infer() throws Exception {
    // Get the API key for invoking Tiyaro API
    var apiKey = System.getenv("TIYARO_API_KEY");
    if (null == apiKey || apiKey.trim().isEmpty()) {
      throw new IllegalArgumentException(
          "Please set TIYARO_API_KEY environment variable. You can generate your API key from here - https://console.tiyaro.ai/apikeys");
    }
    // API endpoint
    var url = "https://api.tiyaro.ai/v1/ent/aws/1/rekognition_compare_faces";
    // Convert binary image to base64
    var img1Path = "testdata/tom1.jpg";
    var b64Img1 = imageToBase64(img1Path);
    var img2Path = "testdata/tom1.jpg";
    var b64Img2 = imageToBase64(img2Path);
    var client = HttpClient.newHttpClient();

    var payload = new JSONObject();
    payload.put("SourceImage", Collections.singletonMap("Bytes", b64Img1));
    payload.put("TargetImage", Collections.singletonMap("Bytes", b64Img2));

    var request =
        HttpRequest.newBuilder(URI.create(url))
            .setHeader("accept", "*/*")
            .setHeader("content-type", "application/json")
            .setHeader("authorization", String.format("Bearer %s", apiKey))
            .POST(BodyPublishers.ofString(payload.toString()))
            .build();

    System.out.println(String.format("Making inference request to: %s", url));
    var response = client.send(request, BodyHandlers.ofString());
    System.out.println(String.format("Response Status: %s", response.statusCode()));
    System.out.println(
        String.format("Response Body: \n%s", new JSONObject(response.body()).toString(4)));
    assert 200 == response.statusCode();
  }

  public static void main(String[] args) throws Exception {
    LocalImageInput demo = new LocalImageInput();
    demo.infer();
  }
}
